test/test_common/environment.cc

```diff
@@ -206,4 +211,15 @@ void TestEnvironment::exec(const std::vector<std::string>& args) {
     RELEASE_ASSERT(false);
   }
 }
+
+std::string TestEnvironment::writeStringToFileForTest(const std::string& filename,
```

Comment:
Can you `grep -r ofstream test/` and replace the various sites where this is done manually with your new helper?